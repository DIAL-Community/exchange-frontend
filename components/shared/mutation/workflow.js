import { gql } from '@apollo/client'

export const UPDATE_WORKFLOW_BUILDING_BLOCKS = gql`
  mutation UpdateWorkflowBuildingBlocks (
    $buildingBlockSlugs: [String!]!
    $slug: String!
  ) {
    updateWorkflowBuildingBlocks (
      buildingBlockSlugs: $buildingBlockSlugs
      slug: $slug
    ) {
      workflow {
        id
        name
        slug
        buildingBlocks {
          id
          slug
          name
          imageFile
          maturity
        }
      }
      errors
    }
  }
`

export const CREATE_WORKFLOW = gql`
  mutation CreateWorkflow (
    $name: String!
    $slug: String!
    $imageFile: Upload
    $description: String!
  ) {
    createWorkflow (
      name: $name
      slug: $slug
      imageFile: $imageFile
      description: $description
    ) {
      workflow {
        id
        name
        slug
        imageFile
        workflowDescription {
          id
          description
          locale
        }
      }
      errors
    }
  }
`

export const DELETE_WORKFLOW = gql`
  mutation DeleteWorkflow($id: ID!) {
    deleteWorkflow(id: $id) {
      workflow {
       id
       slug
       name
      }
      errors
    }
  }
`

