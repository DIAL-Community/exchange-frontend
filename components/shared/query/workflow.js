import { gql } from '@apollo/client'

export const WORKFLOW_SEARCH_QUERY = gql`
  query Workflows($search: String) {
    workflows(search: $search) {
      id
      name
      slug
    }
  }
`

export const WORKFLOW_PAGINATION_ATTRIBUTES_QUERY = gql`
  query PaginationAttributeWorkflow(
    $search: String
    $sdgs: [String!]
    $useCases: [String!]
  ) {
    paginationAttributeWorkflow(
      search: $search
      sdgs: $sdgs
      useCases: $useCases
    ) {
      totalCount
    }
  }
`

export const PAGINATED_WORKFLOWS_QUERY = gql`
  query PaginatedWorkflows(
    $search: String
    $sdgs: [String!]
    $useCases: [String!]
    $limit: Int!
    $offset: Int!
  ) {
    paginatedWorkflows(
      search: $search
      sdgs: $sdgs
      useCases: $useCases
      offsetAttributes: { limit: $limit, offset: $offset }
    ) {
      id
      name
      slug
      imageFile
      parsedDescription
      workflowDescription {
        id
        description
        locale
      }
      buildingBlocks {
        id
      }
      useCases {
        id
      }
    }
  }
`

export const WORKFLOW_DETAIL_QUERY = gql`
  query Workflow($slug: String!) {
    workflow(slug: $slug) {
      id
      name
      slug
      imageFile
      workflowDescription {
        id
        description
        locale
      }
      useCases {
        id
        slug
        name
        maturity
        imageFile
      }
      buildingBlocks {
        id
        name
        slug
        maturity
        imageFile
      }
    }
  }
`
