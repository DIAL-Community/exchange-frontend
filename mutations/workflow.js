import { gql } from '@apollo/client'

export const UPDATE_WORKFLOW_BUILDING_BLOCKS = gql`
      mutation UpdateWorkflowBuildingBlocks (
        $buildingBlocksSlugs: [String!]!
        $slug: String!
        ) {
          updateWorkflowBuildingBlocks (
            buildingBlocksSlugs: $buildingBlocksSlugs
            slug: $slug
          ) {
            workflow {
              slug
              buildingBlocks {
                slug
                name
                imageFile
                maturity
              }
            }
            errors
          }
      }
`
