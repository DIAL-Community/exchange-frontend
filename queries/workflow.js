import { gql } from '@apollo/client'

export const WORKFLOW_SEARCH_QUERY = gql`
  query Workflows($search: String!) {
    workflows(search: $search) {
      id
      name
      slug
    }
  }
`
