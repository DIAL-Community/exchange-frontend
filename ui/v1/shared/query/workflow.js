import { gql } from '@apollo/client'

export const WORKFLOW_SEARCH_QUERY = gql`
  query Workflows($search: String) {
    workflows(search: $search) {
      id
      name
      slug
    }
  }
`

export const WORKFLOW_PAGINATION_ATTRIBUTES_QUERY = gql`
  query PaginationAttributeWorkflow(
    $search: String
  ) {
    paginationAttributeWorkflow(
      search: $search
    ) {
      totalCount
    }
  }
`

export const PAGINATED_WORKFLOWS_QUERY = gql`
  query PaginatedWorkflows(
    $search: String
    $limit: Int!
    $offset: Int!
  ) {
    paginatedWorkflows(
      search: $search
      offsetAttributes: { limit: $limit, offset: $offset }
    ) {
      id
      name
      slug
      imageFile
      workflowDescription {
        id
        description
        locale
      }
      buildingBlocks {
        id
      }
      useCases {
        id
      }
    }
  }
`

export const WORKFLOW_DETAIL_QUERY = gql`
  query Workflow($slug: String!) {
    workflow(slug: $slug) {
      id
      name
      slug
      imageFile
      workflowDescription {
        description
        locale
      }
      useCases {
        slug
        name
        maturity
        imageFile
      }
      buildingBlocks {
        name
        slug
        maturity
        imageFile
      }
    }
  }
`
