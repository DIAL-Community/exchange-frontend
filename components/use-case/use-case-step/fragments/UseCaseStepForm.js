import React, { useCallback, useContext, useState } from 'react'
import { useRouter } from 'next/router'
import { Controller, useForm } from 'react-hook-form'
import { FaSpinner } from 'react-icons/fa6'
import { useIntl } from 'react-intl'
import { useMutation } from '@apollo/client'
import { ToastContext } from '../../../../lib/ToastContext'
import { HtmlEditor } from '../../../shared/form/HtmlEditor'
import Input from '../../../shared/form/Input'
import ValidationError from '../../../shared/form/ValidationError'
import { CREATE_USE_CASE_STEP } from '../../../shared/mutation/useCaseStep'

const UseCaseStepForm = React.memo(({ useCaseStep, useCase }) => {
  const { formatMessage } = useIntl()
  const format = useCallback((id, values) => formatMessage({ id }, values), [formatMessage])

  const router = useRouter()

  const [mutating, setMutating] = useState(false)
  const [reverting, setReverting] = useState(false)

  const { showSuccessMessage, showFailureMessage } = useContext(ToastContext)
  const { locale } = useRouter()

  const [updateUseCaseStep, { reset }] = useMutation(CREATE_USE_CASE_STEP, {
    onError: () => {
      showFailureMessage(format('toast.submit.failure', { entity: format('ui.useCaseStep.label') }))
      setMutating(false)
      reset()
    },
    onCompleted: (data) => {
      const { createUseCaseStep: response } = data
      if (response?.useCaseStep && response?.errors?.length === 0) {
        setMutating(false)
        showSuccessMessage(
          format('toast.submit.success', { entity: format('ui.useCaseStep.label') }),
          () => router.push(
            `/${locale}` +
            `/use-cases/${data.createUseCaseStep.useCaseStep.useCase.slug}` +
            `/use-case-steps/${data.createUseCaseStep.useCaseStep.slug}`
          )
        )
      } else {
        showFailureMessage(format('toast.submit.failure', { entity: format('ui.useCaseStep.label') }))
        setMutating(false)
        reset()
      }
    }
  })

  const { handleSubmit, register, control, formState: { errors } } = useForm({
    mode: 'onSubmit',
    reValidateMode: 'onChange',
    shouldUnregister: true,
    defaultValues: {
      name: useCaseStep?.name,
      stepNumber: useCaseStep?.stepNumber,
      description: useCaseStep?.useCaseStepDescription?.description
    }
  })

  const slug = useCaseStep?.slug ?? ''
  const useCaseId = parseInt(useCase?.id)

  const doUpsert = async (data) => {
    setMutating(true)

    const stepNumber = parseInt(data.stepNumber)

    const { name, description } = data

    const variables = {
      name,
      slug,
      stepNumber,
      description,
      useCaseId
    }
    updateUseCaseStep({
      variables,
      context: {
        headers: {
          'Accept-Language': locale
        }
      }
    })
  }

  const cancelForm = () => {
    setReverting(true)
    router.push(
      `/${locale}` +
      `/use-cases/${useCase?.slug}` +
      `/${slug && 'use-case-steps/' + slug}`
    )
  }

  return (
    <form onSubmit={handleSubmit(doUpsert)}>
      <div className='px-4 lg:px-0 py-4 lg:py-6'>
        <div className='flex flex-col gap-y-6 text-sm'>
          <div className='text-2xl font-semibold text-dial-sapphire'>
            {useCaseStep?.slug
              ? format('app.editEntity', { entity: useCaseStep?.name })
              : `${format('app.createNew')} ${format('use-case-step.label')}`
            }
          </div>
          <div className='form-field-wrapper'>
            <label className='required-field text-dial-blueberry' htmlFor='name'>
              {format('use-case-step.name')}
            </label>
            <Input
              {...register('name', { required: format('validation.required') })}
              id='name'
              placeholder={format('use-case-step.name')}
              isInvalid={errors.name}
            />
            {errors.name && <ValidationError value={errors.name?.message} />}
          </div>
          <div className='form-field-wrapper'>
            <label className='required-field text-dial-blueberry' htmlFor='stepNumber'>
              {format('use-case-step.step-number')}
            </label>
            <Input
              type='number'
              {...register('stepNumber', { required: format('validation.required') })}
              id='stepNumber'
              placeholder={format('use-case-step.step-number')}
              isInvalid={errors.stepNumber}
            />
            {errors.stepNumber && <ValidationError value={errors.stepNumber?.message} />}
          </div>
          <div className='flex flex-col gap-y-2'>
            <label id='description' className='required-field text-dial-blueberry'>
              {format('use-case-step.description')}
            </label>
            <Controller
              name='description'
              control={control}
              render={({ field: { value, onChange } }) => (
                <HtmlEditor
                  labelledBy='description'
                  onChange={onChange}
                  initialContent={value}
                  placeholder={format('use-case-step.description')}
                />
              )}
              rules={{ required: format('validation.required') }}
            />
            {errors.description && <ValidationError value={errors.description?.message} />}
          </div>
          <div className='flex flex-wrap text-base mt-6 gap-3'>
            <button
              type='submit'
              className='submit-button'
              disabled={mutating || reverting}
            >
              {`${format('app.submit')} ${format('use-case-step.label')}`}
              {mutating && <FaSpinner className='spinner ml-3' />}
            </button>
            <button
              type='button'
              className='cancel-button'
              disabled={mutating || reverting}
              onClick={cancelForm}
            >
              {format('app.cancel')}
              {reverting && <FaSpinner className='spinner ml-3' />}
            </button>
          </div>
          {useCase.markdownUrl &&
            <div className='text-sm italic text-red-500 -mt-3'>
              {format('useCaseStep.markdownWarning')}
            </div>
          }
        </div>
      </div>
    </form>
  )
})

UseCaseStepForm.displayName = 'UseCaseStepForm'

export default UseCaseStepForm
