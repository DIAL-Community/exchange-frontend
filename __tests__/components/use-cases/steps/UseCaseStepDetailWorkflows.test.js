import { fireEvent, screen } from '@testing-library/react'
import { mockRouterImplementation, mockSessionImplementation, render, waitForAllEffectsAndSelectToLoad } from '../../../test-utils'
import CustomMockedProvider, { generateMockApolloData } from '../../../utils/CustomMockedProvider'
import { WORKFLOW_SEARCH_QUERY } from '../../../../queries/workflow'
import UseCaseStepDetailWorkflows from '../../../../components/use-cases/steps/UseCaseStepDetailWorkflows'
import { workflows } from './data/UseCaseStepDetailWorkflows'
import { useCaseStep } from './data/UseCaseStepForm'

jest.mock('next/dist/client/router')
jest.mock('next-auth/client')

describe('Unit test for the UseCaseStepDetailWorkflows component.', () => {
  const EDIT_BUTTON_TEST_ID = 'edit-button'
  const CANCEL_BUTTON_TEST_ID = 'cancel-button'
  const WORKFLOW_SEARCH_TEST_ID = 'workflow-search'
  const WORKFLOW_SEARCH_OPTION_LABEL = 'Another Workflow'
  const USECASE_TEST_WORKFLOW_LABEL = 'Test Workflow'
  const PILL_TEST_ID = 'pill'
  const PILL_REMOVE_BUTTON_TEST_ID = 'remove-button'
  const mockWorkflows = generateMockApolloData(WORKFLOW_SEARCH_QUERY, { search: '' }, null, workflows)

  beforeAll(() => {
    mockRouterImplementation()
    mockSessionImplementation()
  })

  test('Should match snapshot - without edit permission.', () => {
    const { container } = render(
      <CustomMockedProvider mocks={[mockWorkflows]}>
        <UseCaseStepDetailWorkflows
          canEdit={false}
          useCaseStep={useCaseStep}
        />
      </CustomMockedProvider>
    )
    expect(container).toMatchSnapshot()
  })

  test('Should match snapshot - with edit permission.', () => {
    const { container } = render(
      <CustomMockedProvider mocks={[mockWorkflows]}>
        <UseCaseStepDetailWorkflows
          canEdit={false}
          useCaseStep={useCaseStep}
        />
      </CustomMockedProvider>
    )
    expect(container).toMatchSnapshot()
  })

  test('Should match snapshot - with open editable section', async () => {
    const { container, getByTestId } = render(
      <CustomMockedProvider mocks={[mockWorkflows]}>
        <UseCaseStepDetailWorkflows
          canEdit={true}
          useCaseStep={useCaseStep}
        />
      </CustomMockedProvider>
    )
    fireEvent.click(getByTestId(EDIT_BUTTON_TEST_ID))
    await waitForAllEffectsAndSelectToLoad(container)
    expect(container).toMatchSnapshot()
  })

  test('Should remove a pill', async () => {
    const { container, getByTestId } = render(
      <CustomMockedProvider mocks={[mockWorkflows]}>
        <UseCaseStepDetailWorkflows
          canEdit={true}
          useCaseStep={useCaseStep}
        />
      </CustomMockedProvider>
    )
    fireEvent.click(getByTestId(EDIT_BUTTON_TEST_ID))
    await waitForAllEffectsAndSelectToLoad(container)
    fireEvent.click(getByTestId(PILL_REMOVE_BUTTON_TEST_ID))
    expect(screen.queryByTestId(PILL_TEST_ID)).toBeNull()
    expect(container).toMatchSnapshot()
  })

  test('Should add a pill and revert changes on "Cancel" button click', async () => {
    const { container, getByTestId, getByText } = render(
      <CustomMockedProvider mocks={[mockWorkflows]}>
        <UseCaseStepDetailWorkflows
          canEdit={true}
          useCaseStep={useCaseStep}
        />
      </CustomMockedProvider>
    )
    fireEvent.click(getByTestId(EDIT_BUTTON_TEST_ID))
    await waitForAllEffectsAndSelectToLoad(container)

    fireEvent.keyDown(getByTestId(WORKFLOW_SEARCH_TEST_ID).childNodes[1], { key: 'ArrowDown' })
    await screen.findByText(WORKFLOW_SEARCH_OPTION_LABEL)
    fireEvent.click(getByText(WORKFLOW_SEARCH_OPTION_LABEL))
    expect(screen.queryAllByTestId(PILL_TEST_ID)).toHaveLength(2)
    expect(container).toMatchSnapshot()

    fireEvent.click(getByTestId(CANCEL_BUTTON_TEST_ID))
    expect(screen.queryByText(USECASE_TEST_WORKFLOW_LABEL)).toBeInTheDocument()
    expect(screen.queryByText(WORKFLOW_SEARCH_OPTION_LABEL)).not.toBeInTheDocument()

    fireEvent.click(getByTestId(EDIT_BUTTON_TEST_ID))
    await waitForAllEffectsAndSelectToLoad(container)

    expect(screen.queryByText(USECASE_TEST_WORKFLOW_LABEL)).toBeInTheDocument()
    expect(screen.queryByText(WORKFLOW_SEARCH_OPTION_LABEL)).not.toBeInTheDocument()
    expect(screen.queryAllByTestId(PILL_TEST_ID)).toHaveLength(1)
  })
})
