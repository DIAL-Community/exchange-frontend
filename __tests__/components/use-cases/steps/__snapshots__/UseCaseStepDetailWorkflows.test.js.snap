// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit test for the UseCaseStepDetailWorkflows component. Should add a pill and revert changes on "Cancel" button click 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="mt-12"
    data-testid="editable-section"
  >
    <div
      class="flex justify-between items-center mb-3"
    >
      <div
        class="card-title text-dial-gray-dark"
      >
        Workflows
      </div>
      <div
        class="gap-3"
      />
    </div>
    <div
      class="bg-edit p-6"
    >
      <p
        class="card-title text-dial-blue mb-3"
      >
        Assign
         
        Workflows
      </p>
      <label
        class="flex flex-col gap-y-2 mb-2"
        data-testid="workflow-search"
      >
        Search and Assign Workflows
        <div
          class=" css-l45kjq-container"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-:r2:-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class="react-select__control css-19jej77-control"
          >
            <div
              class="react-select__value-container css-3tzzl5-ValueContainer"
            >
              <div
                class="react-select__placeholder css-19jynah-placeholder"
                id="react-select-:r2:-placeholder"
              >
                Type to search...
              </div>
              <div
                class="react-select__input-container css-6j8wv5-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-describedby="react-select-:r2:-placeholder"
                  aria-expanded="false"
                  aria-haspopup="true"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class="react-select__input"
                  id="async-select-id-undefined"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 0; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class="react-select__indicator-separator css-1979yuk-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class="react-select__indicator react-select__dropdown-indicator css-1hmmjq6-indicatorContainer"
              >
                <svg
                  class="text-2xl"
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </label>
      <div
        class="flex flex-wrap gap-3 mt-5"
      >
        <div
          class="inline-flex items-center px-2 py-1 rounded-md bg-dial-blue-light text-lg text-dial-gray-dark"
          data-testid="pill"
        >
          Test Workflow
          <svg
            class="ml-3 text-2xl cursor-pointer"
            data-testid="remove-button"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </div>
        <div
          class="inline-flex items-center px-2 py-1 rounded-md bg-dial-blue-light text-lg text-dial-gray-dark"
          data-testid="pill"
        >
          Another Workflow
          <svg
            class="ml-3 text-2xl cursor-pointer"
            data-testid="remove-button"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </div>
      </div>
      <div
        class="flex justify-end mt-8 gap-3 text-xl"
      >
        <button
          class="submit-button"
          data-testid="submit-button"
          type="submit"
        >
          Submit
        </button>
        <button
          class="cancel-button"
          data-testid="cancel-button"
          type="button"
        >
          Cancel
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Unit test for the UseCaseStepDetailWorkflows component. Should match snapshot - with edit permission. 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="mt-12"
    data-testid="editable-section"
  >
    <div
      class="flex justify-between items-center mb-3"
    >
      <div
        class="card-title text-dial-gray-dark"
      >
        Workflows
      </div>
      <div
        class="gap-3"
      />
    </div>
    <div
      class="flex flex-col gap-2"
    >
      <div
        class="
  border-3 cursor-pointer
  border-transparent hover:border-dial-yellow
  text-workflow hover:text-dial-yellow
"
      >
        <div
          class="bg-white border border-dial-gray hover:border-transparent card-drop-shadow"
        >
          <div
            class="grid grid-cols-12 gap-x-4 py-4 px-4"
          >
            <div
              class="col-span-12 
   whitespace-nowrap text-ellipsis overflow-hidden my-auto
 text-base font-semibold"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2720%27%20height=%2720%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="Logo for: Test Workflow"
                  class="workflow-filter"
                  data-nimg="intrinsic"
                  data-tip="Workflow: Test Workflow"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
              <span
                class="ml-4"
              >
                Test Workflow
              </span>
            </div>
            <div
              class="
                      hidden lg:block
                      lg:col-span-4 text-base text-use-case 
   whitespace-nowrap text-ellipsis overflow-hidden my-auto

                    "
            />
            <div
              class="
                      hidden lg:block
                      lg:col-span-4 text-base text-building-block 
   whitespace-nowrap text-ellipsis overflow-hidden my-auto

                    "
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Unit test for the UseCaseStepDetailWorkflows component. Should match snapshot - with open editable section 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="mt-12"
    data-testid="editable-section"
  >
    <div
      class="flex justify-between items-center mb-3"
    >
      <div
        class="card-title text-dial-gray-dark"
      >
        Workflows
      </div>
      <div
        class="gap-3"
      />
    </div>
    <div
      class="bg-edit p-6"
    >
      <p
        class="card-title text-dial-blue mb-3"
      >
        Assign
         
        Workflows
      </p>
      <label
        class="flex flex-col gap-y-2 mb-2"
        data-testid="workflow-search"
      >
        Search and Assign Workflows
        <div
          class=" css-l45kjq-container"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-:r0:-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class="react-select__control css-19jej77-control"
          >
            <div
              class="react-select__value-container css-3tzzl5-ValueContainer"
            >
              <div
                class="react-select__placeholder css-19jynah-placeholder"
                id="react-select-:r0:-placeholder"
              >
                Type to search...
              </div>
              <div
                class="react-select__input-container css-6j8wv5-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-describedby="react-select-:r0:-placeholder"
                  aria-expanded="false"
                  aria-haspopup="true"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class="react-select__input"
                  id="async-select-id-undefined"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class="react-select__indicator-separator css-1979yuk-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class="react-select__indicator react-select__dropdown-indicator css-1hmmjq6-indicatorContainer"
              >
                <svg
                  class="text-2xl"
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </label>
      <div
        class="flex flex-wrap gap-3 mt-5"
      >
        <div
          class="inline-flex items-center px-2 py-1 rounded-md bg-dial-blue-light text-lg text-dial-gray-dark"
          data-testid="pill"
        >
          Test Workflow
          <svg
            class="ml-3 text-2xl cursor-pointer"
            data-testid="remove-button"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </div>
      </div>
      <div
        class="flex justify-end mt-8 gap-3 text-xl"
      >
        <button
          class="submit-button"
          data-testid="submit-button"
          disabled=""
          type="submit"
        >
          Submit
        </button>
        <button
          class="cancel-button"
          data-testid="cancel-button"
          type="button"
        >
          Cancel
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Unit test for the UseCaseStepDetailWorkflows component. Should match snapshot - without edit permission. 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="mt-12"
    data-testid="editable-section"
  >
    <div
      class="flex justify-between items-center mb-3"
    >
      <div
        class="card-title text-dial-gray-dark"
      >
        Workflows
      </div>
      <div
        class="gap-3"
      />
    </div>
    <div
      class="flex flex-col gap-2"
    >
      <div
        class="
  border-3 cursor-pointer
  border-transparent hover:border-dial-yellow
  text-workflow hover:text-dial-yellow
"
      >
        <div
          class="bg-white border border-dial-gray hover:border-transparent card-drop-shadow"
        >
          <div
            class="grid grid-cols-12 gap-x-4 py-4 px-4"
          >
            <div
              class="col-span-12 
   whitespace-nowrap text-ellipsis overflow-hidden my-auto
 text-base font-semibold"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2720%27%20height=%2720%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="Logo for: Test Workflow"
                  class="workflow-filter"
                  data-nimg="intrinsic"
                  data-tip="Workflow: Test Workflow"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
              <span
                class="ml-4"
              >
                Test Workflow
              </span>
            </div>
            <div
              class="
                      hidden lg:block
                      lg:col-span-4 text-base text-use-case 
   whitespace-nowrap text-ellipsis overflow-hidden my-auto

                    "
            />
            <div
              class="
                      hidden lg:block
                      lg:col-span-4 text-base text-building-block 
   whitespace-nowrap text-ellipsis overflow-hidden my-auto

                    "
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Unit test for the UseCaseStepDetailWorkflows component. Should remove a pill 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="mt-12"
    data-testid="editable-section"
  >
    <div
      class="flex justify-between items-center mb-3"
    >
      <div
        class="card-title text-dial-gray-dark"
      >
        Workflows
      </div>
      <div
        class="gap-3"
      />
    </div>
    <div
      class="bg-edit p-6"
    >
      <p
        class="card-title text-dial-blue mb-3"
      >
        Assign
         
        Workflows
      </p>
      <label
        class="flex flex-col gap-y-2 mb-2"
        data-testid="workflow-search"
      >
        Search and Assign Workflows
        <div
          class=" css-l45kjq-container"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-:r1:-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class="react-select__control css-19jej77-control"
          >
            <div
              class="react-select__value-container css-3tzzl5-ValueContainer"
            >
              <div
                class="react-select__placeholder css-19jynah-placeholder"
                id="react-select-:r1:-placeholder"
              >
                Type to search...
              </div>
              <div
                class="react-select__input-container css-6j8wv5-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-describedby="react-select-:r1:-placeholder"
                  aria-expanded="false"
                  aria-haspopup="true"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class="react-select__input"
                  id="async-select-id-undefined"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class="react-select__indicator-separator css-1979yuk-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class="react-select__indicator react-select__dropdown-indicator css-1hmmjq6-indicatorContainer"
              >
                <svg
                  class="text-2xl"
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </label>
      <div
        class="flex flex-wrap gap-3 mt-5"
      />
      <div
        class="flex justify-end mt-8 gap-3 text-xl"
      >
        <button
          class="submit-button"
          data-testid="submit-button"
          type="submit"
        >
          Submit
        </button>
        <button
          class="cancel-button"
          data-testid="cancel-button"
          type="button"
        >
          Cancel
        </button>
      </div>
    </div>
  </div>
</div>
`;
